midend/predication.cpp

```diff
@@ -250,18 +248,12 @@ const IR::Node* Predication::preorder(IR::IfStatement* statement) {
         visit(statement->ifFalse);
         rv->push_back(statement->ifFalse);
     }
-    rv->push_back(blocks.back());
-    blocks.pop_back();
-    blocks.push_back(new IR::BlockStatement);
-    for (auto exprName : orderedNames) {
-        if (!isAssignmentPushed[liveAssignments[exprName]]) {
-            rv->push_back(liveAssignments[exprName]);
-        }
-        isAssignmentPushed[liveAssignments[exprName]] = false;
+    // Push assignments which are correctly aranged in liveAssigns vector on rv block.
+    for (auto it : liveAssigns) {
```

Comment:
use vector::insert?

Reply:
vector::insert can't be used in this case because rv is a BlockStatement pointer, not a vector. Function push_back is defined for BlockStatements. The only way insert could be used here is if it would be defined for BlockStatements.